src/manage_sql_report_formats.c

```diff
@@ -3916,6 +3916,59 @@ empty_trashcan_report_formats ()
   return 0;
 }
 
+/**
+ * @brief Change ownership of report formats, for user deletion.
+ *
+ * @param[in]  user       Current owner.
+ * @param[in]  inheritor  New owner.
+ */
+static void
+inherit_report_format_dir (const gchar *report_format_id, user_t user,
+                           user_t inheritor)
+{
+  gchar *user_id, *inheritor_id, *old_dir, *new_dir;
+
+  g_debug ("%s: %s from %llu to %llu", __func__, report_format_id, user,
+           inheritor);
+
+  user_id = user_uuid (user);
+  if (user_id == NULL)
+    {
+      g_warning ("%s: user_id NULL, skipping report format dir", __func__);
+      return;
+    }
+
+  inheritor_id = user_uuid (inheritor);
+  if (inheritor_id == NULL)
+    {
+      g_warning ("%s: inheritor_id NULL, skipping report format dir", __func__);
+      return;
+    }
+
+  old_dir = g_build_filename (GVMD_STATE_DIR,
```

Comment:
This will be wrong for old predefined report formats.  Their owner is changed in migrate_predefined_report_formats, but their files will still be in the old special dir (and not in the owner's report format dir).  I'll fix this in a new PR.